### PR TITLE
plotter: EventPlot: add user-defined color map

### PR DESCRIPTION
**Pull request by mdigiorgio**
Give the possibility to specify a mapping between events and custom colours in
the plot.

User can specify colours as both colour name or using its HEX representation.

Signed-off-by: Michele Di Giorgio <michele.digiorgio@arm.com>

### CODE REVIEW

File: trappy/plotter/EventPlot.py [hunk 1]
```diff
@@ -87,6 +87,16 @@ class EventPlot(AbstractDataPlotter):
...
         :type lanes: list
+
+        :param color_map: A mapping between events and colours
+            ::
+                { "<name1>" : "colour1",
+                  .
+                  .
+                  .
+                  "<nameN>" : "colourN"
+                }
```
Comment (JaviMerino) [inline trappy/plotter/EventPlot.py]: Please document how you specify colors.  If by name, we could point to a table with names (coming from some HTML documentation?).  For hex, maybe put an example `#ff0000` is red or something like that.

**Comment by JaviMerino** [comment]
Can you add the example to the InteractivePlotter notebook?  Maybe just add it to one of the EventPlot examples.

File: trappy/plotter/EventPlot.py (hunk 1 above)
Comment (mdigiorgio) [inline trappy/plotter/EventPlot.py]: Sure :smile:

**Comment by mdigiorgio** [comment]
:+1:

**Pull request merged**